Library/Homebrew/formula.rb

```diff
@@ -494,7 +494,16 @@ def python(options={}, &block)
 
   # an array of all {Formula} names
   def self.names
-    Dir["#{HOMEBREW_LIBRARY}/Formula/*.rb"].map{ |f| File.basename f, '.rb' }.sort
+    core = Dir["#{HOMEBREW_LIBRARY}/Formula/*.rb"].map{ |f| File.basename f, '.rb' }.sort
+    taps = Dir["#{HOMEBREW_LIBRARY}/Taps/*/*/"].reduce([]) do |formulae, tap|
+      Pathname.new(tap).find_formula do |f|
+        if f.to_s =~ HOMEBREW_TAP_PATH_REGEX
+           formulae << "#$1/#{$2.gsub(/^homebrew-/, "")}/#{f.basename(".rb")}"
+        end
+      end
+      formulae
+    end.sort
+    core+taps
   end
```

Comment:
Well, if we return all the names of the formulae across all taps then it'll be consistent with the current behaviour. It doesn't really matter hugely if some of those names are duplicated across taps for this first pass, I think.